pkg/machines/app.jsx

```diff
@@ -18,21 +18,71 @@
  */
 import React from 'react';
 import PropTypes from 'prop-types';
+import { ToastNotificationList, ToastNotification } from 'patternfly-react';
 
 import HostVmsList from "./hostvmslist.jsx";
 import { StoragePoolList } from "./components/storagePools/storagePoolList.jsx";
 import { NetworkList } from "./components/networks/networkList.jsx";
 import LibvirtSlate from "./components/libvirtSlate.jsx";
 import { CreateVmAction } from "./components/create-vm-dialog/createVmDialog.jsx";
 import { AggregateStatusCards } from "./components/aggregateStatusCards.jsx";
+import { InlineNotification } from './components/notification/inlineNotification.jsx';
 
 class App extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            activeTab: 1
+            activeTab: 1,
+            notifications: [],
+            /* Dictionary with keys being a resource's UUID and values the number of active error notifications for that resource */
+            resourceHasError: {},
         };
         this.changeActiveList = this.changeActiveList.bind(this);
+        this.onAddErrorNotification = this.onAddErrorNotification.bind(this);
+        this.onDismissErrorNotification = this.onDismissErrorNotification.bind(this);
+    }
+
+    /*
+     * Adds a new notification object to the notificationsArray. It also updates
+     * the error count for a specific resource.
+     * @param {object} notification - The notification object to be added to the array.
+     */
+    onAddErrorNotification(notification) {
+        let resourceHasError = Object.assign({}, this.state.resourceHasError);
+
+        if (resourceHasError[notification.resourceId])
+            resourceHasError[notification.resourceId]++;
+        else
+            resourceHasError[notification.resourceId] = 1;
+
+        notification.index = this.state.notifications.length;
+
+        this.setState({
+            notifications: [
+                ...this.state.notifications,
+                notification
+            ],
+            resourceHasError,
+        });
+    }
+
+    /*
+     * Removes the notification with index notificationIndex from the notificationsArray.
+     * It also updates the error count for a specific resource.
+     * @param {int} notificationIndex - Index of the notification to be removed.
+     */
+    onDismissErrorNotification(notificationIndex) {
+        let notificationsArray = this.state.notifications.concat();
+        const index = notificationsArray.findIndex(current => current.index == notificationIndex);
+
+        if (index !== -1) {
+            let resourceHasError = Object.assign({}, this.state.resourceHasError);
+
+            resourceHasError[notificationsArray[index].resourceId]--;
+
+            notificationsArray.splice(index, 1);
```

Comment:
This is also wrong, as the index at the time of creation (`current.index`) is different from the *current* index. Moving to a counter and a dictionary will avoid that and also simplify the lookup.

Reply:
Done